tests/Types/VarDateTimeImmutableTypeTest.php

```diff
@@ -40,11 +40,11 @@ public function testConvertsDateTimeImmutableInstanceToDatabaseValue(): void
 
         $date->expects(self::once())
             ->method('format')
-            ->with('Y-m-d H:i:s')
-            ->willReturn('2016-01-01 15:58:59');
+            ->with('Y-m-d H:i:s.u')
+            ->willReturn('2016-01-01 15:58:59.000000');
```

Comment:
Changes in existing tests often identify breaking changes.

Reply:
Even in a mock configuration?

Side note: Do we actually _need_ this mock? 🫣

Reply:
In this case, do we need this change?

Reply:
There is definitely a change to behaviour - the format string now has a microseconds specifier and the string presented to the database has six decimal digits. The functional tests test all precisions of datetime (including zero) against all column precisions to ensure the behaviour is as expected.  

Whether or not we use the mock, there will need to be a change to the assertion in the test to reflect the additional decimal digits in the datetime string.